src/QuiqChatClient.js

```diff
@@ -2,22 +2,25 @@
 import * as API from './apiCalls';
 import {setGlobals, checkRequiredSettings} from './globals';
 import {connectSocket, disconnectSocket} from './websockets';
-import type {AtmosphereMessage, Message, ApiError, UserEventTypes} from './types';
+import type {AtmosphereMessage, TextMessage, ApiError, UserEventTypes, Event} from './types';
 import {MessageTypes, quiqChatContinuationCookie} from './appConstants';
 import {set, get} from 'js-cookie';
-import {differenceBy, last} from 'lodash';
+import {differenceBy, last, partition} from 'lodash';
+import {sortByTimestamp} from './utils';
 
-const getConversationMessages = async () => {
+const getConversation = async (): Promise<{events: Array<Event>, messages: Array<TextMessage>}> => {
   const conversation = await API.fetchConversation();
-  return conversation.messages.filter(
-    m =>
-      m.type === MessageTypes.TEXT &&
-      !m.text.trim().includes('Quiq Welcome Form Customer Submission'),
+  const partitionedConversation = partition(conversation.messages, {type: MessageTypes.TEXT});
+  // NOTE: For backwards compatibility, we must filter out 'Quiq Welcome Form Customer Submission' textMessages
+  const messages = partitionedConversation[0].filter(
+    m => !m.text.trim().includes('Quiq Welcome Form Customer Submission'),
```

Comment:
Responding to Fred's comment here.  Pasted below for context.

Fred: It should still be there of backwards compatibility though shouldn’t it?

---

This isn't really a backwards compatibility issue for a couple reasons
1. This is a pure client-side hack.  The only conversations that would still need this hack would be those conversations that are active while the new version of webchat goes to production.  Any new chats after that won't need this code

2.  It is true that those few conversation that land in the above category would be 'broken' by this, however, the effect of removing this code is just that those customers would see a copy of their own submitted WelcomeForm data, which isn't really that detrimental (we actually had it this way in production for quite some time before we chose to change it).

3. For those end users that would be affected by this, they would need to submit their welcome form, then refresh their page and receive the new version of webchat during that conversation.  This would only affect them for the duration of that conversation.